### PR TITLE
Qtile: 0.22.1 -> 0.23.0

### LINKED ISSUE #252769: Build failure: python-3.10-qtile-0.22.1
### Steps To Reproduce

1. run `nixos-rebuild boot --upgrade`

### Build log

[qtile build fail log](https://gist.github.com/Zemrod/2b4dd102246e6643b3a378f426916b4e)

### Additional context

the unstable branch is used

### Notify maintainers

@arjan-s
@kamilchm

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.46, NixOS, 23.11 (Tapir), 23.11pre517472.b85ed9dcbf18`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.0`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by jwijenbergh**
## Description of changes

Updated qtile and deps to upgrade to 0.23.0. Also fixes #252769 in the process

## Things done

See commit descriptions

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by chvp — COMMENTED** [review]
`qtile-extras` doesn't build, don't know how big of an issue that is (it doesn't seem like a matching release has been made).

The commit messages should include the `python3Packages.` prefix.

Otherwise looks good and runs as expected.

**Comment by jwijenbergh** [comment]
We can wait for qtile-extras (the maintainer is on a trip right now), but maybe we should merge this asap as in the current state qtile doesn't build at all

**Review by chvp — APPROVED** [review]

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -11041,9 +11039,7 @@ self: super: with self; {
 
   qtconsole = callPackage ../development/python-modules/qtconsole { };
 
-  qtile = callPackage ../development/python-modules/qtile {
-    wlroots = pkgs.wlroots_0_15;
-  };
+  qtile = callPackage ../development/python-modules/qtile {};
```
Comment (wineee) [inline pkgs/top-level/python-packages.nix:11042]:
```suggestion
  qtile = callPackage ../development/python-modules/qtile {
      wlroots = pkgs.wlroots_0_16;
  };
```
we should keep depending on wlroots_0_* as each new wlroots release comes with so many breaking changes

**Review by wineee — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/qtile/default.nix [hunk 2]
```diff
@@ -48,6 +51,9 @@ buildPythonPackage rec {
       --replace libpango-1.0.so.0 ${pango.out}/lib/libpango-1.0.so.0
     substituteInPlace libqtile/backend/x11/xcursors.py \
       --replace libxcb-cursor.so.0 ${xcbutilcursor.out}/lib/libxcb-cursor.so.0
+    substituteInPlace libqtile/backend/wayland/cffi/build.py \
+        --replace /usr/include/pixman-1 ${pixman.outPath}/include \
```
Comment (wineee) [inline pkgs/development/python-modules/qtile/default.nix:55]:
```suggestion
        --replace /usr/include/pixman-1 ${lib.getDev pixman}/include \
```

File: pkgs/development/python-modules/qtile/default.nix [hunk 3]
```diff
@@ -48,6 +51,9 @@ buildPythonPackage rec {
...
+        --replace /usr/include/libdrm ${libdrm.dev.outPath}/include/libdrm
```
Comment (wineee) [inline pkgs/development/python-modules/qtile/default.nix:56]:
```suggestion
        --replace /usr/include/libdrm ${lib.getDev libdrm}/include/libdrm
```

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (jwijenbergh) [reply pkgs/top-level/python-packages.nix:11042]: Thanks. I thought sway didn't have this but I didn't look correctly.

File: pkgs/development/python-modules/qtile/default.nix (hunk 2 above)
Comment (jwijenbergh) [reply pkgs/development/python-modules/qtile/default.nix]: Neat!

**Review by wineee — APPROVED** [review]

**Comment by arjan-s** [comment]
Result of `nixpkgs-review pr 257211` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.qtile-extras</li>
    <li>python310Packages.qtile-extras.dist</li>
    <li>python311Packages.qtile-extras</li>
    <li>python311Packages.qtile-extras.dist</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python310Packages.pywlroots</li>
    <li>python310Packages.pywlroots.dist</li>
    <li>python310Packages.qtile</li>
    <li>python310Packages.qtile.dist</li>
    <li>python311Packages.pywlroots</li>
    <li>python311Packages.pywlroots.dist</li>
    <li>python311Packages.qtile</li>
    <li>python311Packages.qtile.dist</li>
    <li>qtile</li>
  </ul>
</details>

**Review by arjan-s — APPROVED** [review]
Thanks!

**Comment by arjan-s** [comment]
qtile-extras updated here, reviews/merge appreciated: https://github.com/NixOS/nixpkgs/pull/258101

**Pull request merged**